starknet/README.md

```diff
@@ -0,0 +1,145 @@
+# Starknet Deployment Scripts 🏗️
+
+This directory contains deployment and operational scripts for Axelar contracts on Starknet. The scripts support both online and offline workflows, with hardware wallet integration for secure mainnet deployments.
+
+## 🔧 Setup
+
+### Prerequisites
+
+- Node.js >= 18
+- Starknet.js dependencies
+- For mainnet: Ledger hardware wallet
+
+### Installation
+
+```bash
+npm ci && npm run build
+```
+
+### TypeScript Support
+
+All scripts now use `ts-node` for TypeScript execution:
+- Compatible with existing JavaScript workflows
+- Enhanced type safety and development experience
+- All commands use `ts-node` instead of `node`
+
+### Environment Configuration
+
+Create a `.env` file with the following variables (see `.example.env` for reference):
+
+```bash
+# Starknet Configuration
+STARKNET_PRIVATE_KEY=0x...  # For testnet/devnet only
+STARKNET_ACCOUNT_ADDRESS=0x...
+
+# Network settings
+ENV=testnet  # or mainnet, devnet, stagenet
+CHAIN_NAMES=starknet-sepolia
+```
+
+## 🚀 Core Features
+
+### Dual Workflow Support
+- **Online Mode**: Direct transaction execution (testnet/devnet)
+- **Offline Mode**: Unsigned transaction generation for hardware wallet signing (mainnet)
+
+### Security Model
+- **Testnet/Devnet**: Private key-based signing
+- **Mainnet**: Mandatory offline workflow with Ledger hardware wallets
+
+### Contract Support
+- ✅ Contract deployment and upgrades
+- ✅ Gateway operations (call contract, approve messages, validate messages)
+- ✅ Signer rotation and operatorship management
+- 🔄 Additional contracts (Gas Service, Operators, ITS) - *coming soon*
+
+## 📚 Documentation
+
+### Contract-Specific Guides
+- **[Contract Deployment](./docs/deploy-contract.md)** - Deploy and upgrade contracts
+- **[Gateway Operations](./docs/gateway.md)** - Cross-chain messaging and gateway management
+
+### Workflow Guides
+- **[Offline Signing](./docs/OFFLINE-SIGNING.md)** - Complete guide for mainnet offline workflow
+- **[Key Management](./key-management.md)** - Security guidelines and key management
```

Comment:
It would be nice to have some usage examples. Is the plan to add it to these docs?